orchagent/mirrororch.cpp

```diff
@@ -1115,13 +1134,19 @@ bool MirrorOrch::updateSessionDstMac(const string& name, MirrorEntry& session)
 
     sai_attribute_t attr;
     attr.id = SAI_MIRROR_SESSION_ATTR_DST_MAC_ADDRESS;
-    memcpy(attr.value.mac, session.neighborInfo.mac.getMac(), sizeof(sai_mac_t));
+    if ((gMySwitchType == "voq") && (session.type == MIRROR_SESSION_ERSPAN))
+    {
+         memcpy(attr.value.mac, gMacAddress.getMac(), sizeof(sai_mac_t));
+    } else
+    {
+         memcpy(attr.value.mac, session.neighborInfo.mac.getMac(), sizeof(sai_mac_t));
+    }
 
     sai_status_t status = sai_mirror_api->set_mirror_session_attribute(session.sessionId, &attr);
     if (status != SAI_STATUS_SUCCESS)
     {
         SWSS_LOG_ERROR("Failed to update mirror session %s destination MAC to %s, rv:%d",
-                name.c_str(), session.neighborInfo.mac.to_string().c_str(), status);
+                name.c_str(), sai_serialize_mac(attr.value.mac).c_str(), status);
```

Comment:
Checking if there is any particular reason for using  sai_serialize_mac() here

Reply:
session.neighborInfo.mac is swss:MacAddress  class which has a function to_string. But attr.value.mac is sai_mac_t which is "typedef UINT8   sai_mac_t[6];". So i had to call sai_serialize_mac to convert the mac to string.